core/src/main/java/nl/nn/adapterframework/jdbc/StoredProcedureQuerySender.java

```diff
@@ -83,34 +163,40 @@ public void configure() throws ConfigurationException {
 		if (getQueryTypeEnum() != QueryType.SELECT && getQueryTypeEnum() != QueryType.OTHER) {
 			throw new ConfigurationException("For StoredProcedureSender, queryType can only be 'SELECT' or 'OTHER'");
 		}
-		if (getQueryTypeEnum() == QueryType.SELECT && !getDbmsSupport().isStoredProcedureResultSetSupported()) {
-			throw new ConfigurationException("QueryType SELECT for Stored Procedures is not supported for database " + getDbmsSupport().getDbmsName());
+
+		if (getQueryTypeEnum() == QueryType.OTHER && getOutputFormat() != null) {
+			ConfigurationWarnings.add(this, log, "When querytype is OTHER, the setting for outputFormat is currently ignored.", SuppressKeys.CONFIGURATION_VALIDATION);
 		}
 
 		super.configure();
 
 		// Have to check this after "super.configure()" b/c otherwise the datasource-name is not set
 		// and cannot check DMBS support features.
-		if (!getDbmsSupport().isStoredProceduresSupported()) {
-			throw new ConfigurationException("Stored Procedures are not supported for database " + getDbmsSupport().getDbmsName());
+		if (getQueryTypeEnum() == QueryType.SELECT && !getDbmsSupport().isStoredProcedureResultSetSupported()) {
+			throw new ConfigurationException("QueryType SELECT for Stored Procedures is not supported for database " + getDbmsSupport().getDbmsName());
 		}
 
 		outputParameters = buildOutputParameterMap(getParameterList(), getQuery());
-		if (!outputParameters.isEmpty()) {
-			if (!getDbmsSupport().isStoredProcedureOutParametersSupported()) {
-				throw new ConfigurationException("Stored Procedure OUT parameters are not supported for database " + getDbmsSupport().getDbmsName());
-			}
-		}
-
 		if (isScalar() && outputParameters.size() > 1) {
-			throw new ConfigurationException("When result should be scalar, only a single output can be returned from the stored procedure.");
+			ConfigurationWarnings.add(this, log, "When result should be scalar, only the first output parameter is used. Others are ignored.", SuppressKeys.CONFIGURATION_VALIDATION);
+		}
+		if (getQueryTypeEnum() == QueryType.SELECT && !outputParameters.isEmpty()) {
+			ConfigurationWarnings.add(this, log, "OUT parameters are ignored when QueryType = SELECT", SuppressKeys.CONFIGURATION_VALIDATION);
 		}
 
 		if (!getQuery().matches("(?i)^\\s*(call|exec|\\{\\s*\\?(\\{\\w+\\})?\\s*=\\s*call)\\s+.*")) {
-			throw new ConfigurationException("Stored Procedure query should start with CALL or EXEC SQL statement");
+			ConfigurationWarnings.add(this, log, "Stored Procedure query should start with CALL or EXEC SQL statement", SuppressKeys.CONFIGURATION_VALIDATION);
```

Comment:
Een warning geen exceptie?

Reply:
Ik heb al een paar keer gehad dat er een geldige constructie was die niet voorzien was, of dat ik met iemand iets wou uitproberen om te kijken of het zou werken, maar dat dat niet kon door de exception.

Met een warning kan je toch iets uitproberen maar krijg je wel nog een waarschuwing.

Vandaar dat ik diverse exceptions heb veranderd in warnings.